src/convertToObject.js

```diff
@@ -3,16 +3,25 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ *Function takes a string with styles ([stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * the values of related CSS properties ([test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const newArr = sourceString.split(';')
+    .map(a => a.trim())
+    .filter(a => a.length > 1)
+    .map(a => a.split(':'));
+
+  return newArr.reduce((object, currentValue) => {
+    object[currentValue[0].trim()] = currentValue[1].trim();
+
+    return object;
+  }, {});
```

Comment:
Instead of map, filter, and again map try to do everything in reduce method just in one iteration.